# Cleansing Data Youtube Rewind 2023

#### Import Library

In [1]:
from dotenv import load_dotenv
from pymongo import MongoClient
import os
import pandas as pd

#### Import url dari .env

In [2]:
# Memuat value dari file .env
load_dotenv()

mongodb_url = os.getenv('URL_SANDY')
local_url = os.getenv('URL_LOCAL')

In [3]:
database_name = "youtube_rewind_indonesia"
collection_from = "collect_2023"
collection_name = "cleansing_2023"

#### Import dataset dari MongoDB

In [4]:
# Koneksi ke MongoDB
client = MongoClient(local_url)

# Pilih database dan koleksi
db = client[database_name]
collection = db[collection_from]

# Query data dari koleksi
cursor = collection.find()

# Konversi cursor ke list dan kemudian ke DataFrame
data = list(cursor)
df = pd.DataFrame(data)

# Menutup koneksi
client.close()

In [5]:
df

_id                 channelId      videoId  \
0      665167dbc7b93965e002c573  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
1      665167dbc7b93965e002c574  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
2      665167dbc7b93965e002c575  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
3      665167dbc7b93965e002c576  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
4      665167dbc7b93965e002c577  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
...                         ...                       ...          ...   
38495  66516849c7b93965e0035bd2  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
38496  66516849c7b93965e0035bd3  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
38497  66516849c7b93965e0035bd4  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
38498  66516849c7b93965e0035bd5  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
38499  66516849c7b93965e0035bd6  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   

                                             textDisplay  \
0      Selamat menikmati REWIND INDONESIA di jam 19.4...   
1      KERENN!!!<br>Mulai di menit ke 8 sampai akhir ...   
2                        4.24 itu bukannya bg tanboy ya?   
3                   this is so grazy!!!:&#39;&#39;&#39;(   
4      2024 baru nonton🤚 kukira ada &quot;beras rose ...   
...                                                  ...   
38495                              kereeennn paarrraahgg   
38496                     Kok ga ada om deddy corbuzier?   
38497                                              Keren   
38498  Pas nyanyi maju tak gentar merinding banget si...   
38499                                           TRENDING   

                                            textOriginal authorDisplayName  \
0      Selamat menikmati REWIND INDONESIA di jam 19.4...          @Tim2one   
1      KERENN!!!\nMulai di menit ke 8 sampai akhir fi...  @rosayuliana8209   
2                        4.24 itu bukannya bg tanboy ya?  @user-ly8ck8qd3j   
3                               this is so grazy!!!:'''(  @artikawindy1352   
4        2024 baru nonton🤚 kukira ada "beras rose brand"     @puma_sandewa   
...                                                  ...               ...   
38495                              kereeennn paarrraahgg  @viradkirana7062   
38496                     Kok ga ada om deddy corbuzier?   @ahqproject9474   
38497                                              Keren   @wendyadnan5331   
38498  Pas nyanyi maju tak gentar merinding banget si...    @Bukananakhits   
38499                                           TRENDING         @funnyuID   

                                   authorProfileImageUrl  \
0      https://yt3.ggpht.com/ytc/AIdro_koWWb9JTasq6o_...   
1      https://yt3.ggpht.com/ytc/AIdro_n2ofiv90YFJNp7...   
2      https://yt3.ggpht.com/mjMIIJlCvGr2k8RAyin0OyLT...   
3      https://yt3.ggpht.com/ytc/AIdro_k489dbUjVHiavM...   
4      https://yt3.ggpht.com/ZajaI9dtStey9im1PFBlVEwL...   
...                                                  ...   
38495  https://yt3.ggpht.com/ytc/AIdro_knCgH2a2oqiP8t...   
38496  https://yt3.ggpht.com/ytc/AIdro_mQEuDZgVGTQtcK...   
38497  https://yt3.ggpht.com/ytc/AIdro_nmxR6VNAey7ooY...   
38498  https://yt3.ggpht.com/6waH17VVQRtLSAcuA24gWNZ6...   
38499  https://yt3.ggpht.com/hb1HwS3oPApTVslOWwPvK3sG...   

                              authorChannelUrl  \
0              http://www.youtube.com/@Tim2one   
1      http://www.youtube.com/@rosayuliana8209   
2      http://www.youtube.com/@user-ly8ck8qd3j   
3      http://www.youtube.com/@artikawindy1352   
4         http://www.youtube.com/@puma_sandewa   
...                                        ...   
38495  http://www.youtube.com/@viradkirana7062   
38496   http://www.youtube.com/@ahqproject9474   
38497   http://www.youtube.com/@wendyadnan5331   
38498    http://www.youtube.com/@Bukananakhits   
38499         http://www.youtube.com/@funnyuID   

                             authorChannelId  canRate viewerRating  likeCount  \
0      {'value': 'UCUcBFGAfOzut1x4GSWa6Akg'}     True         none      29687 

#### Menghapus data text yang duplikat

In [6]:
df = df.drop_duplicates(subset=['textOriginal'])

df.shape

(35499, 14)

#### Menghapus data author yang duplikat

In [7]:
df = df.drop_duplicates(subset=['authorChannelUrl'])

df.shape

(34330, 14)

In [8]:
df

_id                 channelId      videoId  \
0      665167dbc7b93965e002c573  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
1      665167dbc7b93965e002c574  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
2      665167dbc7b93965e002c575  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
3      665167dbc7b93965e002c576  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
4      665167dbc7b93965e002c577  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
...                         ...                       ...          ...   
38494  66516849c7b93965e0035bd1  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
38495  66516849c7b93965e0035bd2  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
38496  66516849c7b93965e0035bd3  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
38498  66516849c7b93965e0035bd5  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   
38499  66516849c7b93965e0035bd6  UC-z1ubmS-iHMVr83EEwRS2g  Q5vQawTFJ0I   

                                             textDisplay  \
0      Selamat menikmati REWIND INDONESIA di jam 19.4...   
1      KERENN!!!<br>Mulai di menit ke 8 sampai akhir ...   
2                        4.24 itu bukannya bg tanboy ya?   
3                   this is so grazy!!!:&#39;&#39;&#39;(   
4      2024 baru nonton🤚 kukira ada &quot;beras rose ...   
...                                                  ...   
38494  Keren banget mantap ... Saking kerennya menuru...   
38495                              kereeennn paarrraahgg   
38496                     Kok ga ada om deddy corbuzier?   
38498  Pas nyanyi maju tak gentar merinding banget si...   
38499                                           TRENDING   

                                            textOriginal   authorDisplayName  \
0      Selamat menikmati REWIND INDONESIA di jam 19.4...            @Tim2one   
1      KERENN!!!\nMulai di menit ke 8 sampai akhir fi...    @rosayuliana8209   
2                        4.24 itu bukannya bg tanboy ya?    @user-ly8ck8qd3j   
3                               this is so grazy!!!:'''(    @artikawindy1352   
4        2024 baru nonton🤚 kukira ada "beras rose brand"       @puma_sandewa   
...                                                  ...                 ...   
38494  Keren banget mantap ... Saking kerennya menuru...  @irfadarojatun7247   
38495                              kereeennn paarrraahgg    @viradkirana7062   
38496                     Kok ga ada om deddy corbuzier?     @ahqproject9474   
38498  Pas nyanyi maju tak gentar merinding banget si...      @Bukananakhits   
38499                                           TRENDING           @funnyuID   

                                   authorProfileImageUrl  \
0      https://yt3.ggpht.com/ytc/AIdro_koWWb9JTasq6o_...   
1      https://yt3.ggpht.com/ytc/AIdro_n2ofiv90YFJNp7...   
2      https://yt3.ggpht.com/mjMIIJlCvGr2k8RAyin0OyLT...   
3      https://yt3.ggpht.com/ytc/AIdro_k489dbUjVHiavM...   
4      https://yt3.ggpht.com/ZajaI9dtStey9im1PFBlVEwL...   
...                                                  ...   
38494  https://yt3.ggpht.com/ytc/AIdro_lMF6e1YxbR__1T...   
38495  https://yt3.ggpht.com/ytc/AIdro_knCgH2a2oqiP8t...   
38496  https://yt3.ggpht.com/ytc/AIdro_mQEuDZgVGTQtcK...   
38498  https://yt3.ggpht.com/6waH17VVQRtLSAcuA24gWNZ6...   
38499  https://yt3.ggpht.com/hb1HwS3oPApTVslOWwPvK3sG...   

                                authorChannelUrl  \
0                http://www.youtube.com/@Tim2one   
1        http://www.youtube.com/@rosayuliana8209   
2        http://www.youtube.com/@user-ly8ck8qd3j   
3        http://www.youtube.com/@artikawindy1352   
4           http://www.youtube.com/@puma_sandewa   
...                                          ...   
38494  http://www.youtube.com/@irfadarojatun7247   
38495    http://www.youtube.com/@viradkirana7062   
38496     http://www.youtube.com/@ahqproject9474   
38498      http://www.youtube.com/@Bukananakhits   
38499           http://www.youtube.com/@funnyuID   

                             authorChannelId  canRate viewerRating  likeCount  \
0      {'value': 'UCUcBFGAfOzu

#### Menghapus nilai yang kosong

In [9]:
df = df.dropna()

# Melihat data yang kosong
df.isnull().sum()

_id                      0
channelId                0
videoId                  0
textDisplay              0
textOriginal             0
authorDisplayName        0
authorProfileImageUrl    0
authorChannelUrl         0
authorChannelId          0
canRate                  0
viewerRating             0
likeCount                0
publishedAt              0
updatedAt                0
dtype: int64

Dari yang awalnya berjumlah 38500 row berkurang menjadi 34330 row

#### Mengimpan Clean Comments ke MongoDB Local

In [10]:
# Koneksi ke MongoDB
client = MongoClient(local_url)
db = client[database_name]
collection = db[collection_name]

# Konversi DataFrame ke bentuk dictionary
records = df.to_dict(orient='records')

# Menyisipkan data ke MongoDB
collection.insert_many(records)

print("Data berhasil diimpor ke MongoDB Local.")

Data berhasil diimpor ke MongoDB Local.
